datafusion/core/src/physical_plan/joins/sort_merge_join.rs

```diff
@@ -299,14 +266,7 @@ impl ExecutionPlan for SortMergeJoinExec {
     }
 
     fn maintains_input_order(&self) -> Vec<bool> {
-        match self.join_type {
-            JoinType::Inner => vec![true, true],
```

Comment:
Previously, sort merge join was preserving ordering of both sides. I think this is misleading. There is no way we can preserve ordering of both sides. Hence, I changed this flag to `JoinType::Inner => vec![true, false],`